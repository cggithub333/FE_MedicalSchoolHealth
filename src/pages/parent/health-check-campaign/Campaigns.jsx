import { Grid } from '@mui/material';

import HealthCheckCampaignCard from '../../../components/parent/HealthCheckCampaignCard/HealthCheckCampaignsCard';

import Breadcrumb from '../../../components/magic/Breadcrumb/CustomBreadcrumb';
import CustomTittle from '../../../components/magic/CustomTittle/CustomTitle';

import useLatestHealthCheckCampaign from '../../../hooks/parent/useLatestHealthCheckCampaign';

const Campaigns = () => {

  const { latestHealthCheckCampaign, isLoading } = useLatestHealthCheckCampaign();

  return (
    <div style={{ background: "#e6f8f9", height: "100vh", paddingTop: "20px", paddingBottom: "50px"}}>
      <Grid container>
        <Grid item size={{ xs: 6 }}>
          <Breadcrumb breadcrumbPairs={breadcrumbPairs}/>
        </Grid>
      </Grid>
      <Grid container>
        <Grid item sx={{ marginLeft: "20px", marginTop: "25px" }} size={{ xs: 6}}>
          <CustomTittle title={"Health Check Campaign"}/>
        </Grid>
      </Grid>
      {
        latestHealthCheckCampaign ? 
          (
            <Grid container justifyContent={'center'} sx={styleCardWrapper}>
              <Grid item sx={{ xs: 12 }}>
                <HealthCheckCampaignCard  latestHealthCheckCampaign={latestHealthCheckCampaign}
                                          isLoading={isLoading}
                />
              </Grid>
            </Grid>
          )
          :
          (
            <div style={{
              display: 'flex',
              justifyContent: 'center',
              marginTop: "30px",
              fontSize: "20px",
              fontStyle: "italic"
            }}>There is no ongoing health check campaign!</div>
          )
      }
    </div>
  )
}

const breadcrumbPairs = [
  {
    title: 'Dashboard',
    link: '/parent/dashboard'
  },
  {
    title: 'Campaigns',
    // link: '/parent/vaccination-campaign/campaigns'
  }
]

const styleCardWrapper = {
  marginTop: "30px",
  marginBottom: "100px"
}

export default Campaigns;