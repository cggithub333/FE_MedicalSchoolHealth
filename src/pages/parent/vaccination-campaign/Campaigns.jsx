import Breadcrumb from '../../../components/magic/Breadcrumb/CustomBreadcrumb';
import { Grid, CircularProgress } from "@mui/material";
import VaccinationCampaignCard from "../../../components/parent/VaccinationCampaignCard/VaccinationCampaignCard";
import CustomTittle from '../../../components/magic/CustomTittle/CustomTitle';
import useLatestVaccinationCampaign from '../../../hooks/parent/useLatestVaccinationCampaign';

const Campaigns = () => {
  const { latestVaccinationCampaign, isLoading } = useLatestVaccinationCampaign();

  return (
    <div style={{ background: "#E6F8F9", height: "100vh", paddingBottom: "50px", paddingTop: "10px" }}>
      <Grid container>
        <Grid item size={{ xs: 6 }}>
          <Breadcrumb breadcrumbPairs={breadcrumbPairs} />
        </Grid>
      </Grid>  
      <Grid container>
        <Grid item sx={{ marginLeft: "20px", marginTop: "25px" }} size={{ xs: 6 }}>
          <CustomTittle title={"Vaccination Campaign"} />
        </Grid>
      </Grid>    
      <Grid container justifyContent={'center'} marginTop="35px">
        {isLoading ? (
          <CircularProgress />
        ) : latestVaccinationCampaign ? (
            <VaccinationCampaignCard latestVaccinationCampaign={latestVaccinationCampaign} />
        ) : null}
      </Grid>
    </div>
  )
}

const breadcrumbPairs = [
  {
    title: 'Dashboard',
    link: '/parent/dashboard'
  },
  {
    title: 'Campaigns',
    // link: '/parent/vaccination-campaign/campaigns'
  }
]

export default Campaigns;
