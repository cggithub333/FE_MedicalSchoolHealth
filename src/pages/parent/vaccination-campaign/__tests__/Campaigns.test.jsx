
import React from 'react';
import { render, screen } from '@testing-library/react';
import Campaigns from '../Campaigns';

// Mock child components to isolate Campaigns
jest.mock('@components/parent/VaccinationCampaign/LatestCampaign/LatestVaccinationCampaign', () => () => <div data-testid="LatestVaccinationCampaign" />);
jest.mock('@components/magic/Breadcrumb/CustomBreadcrumb', () => ({ breadcrumbPairs }) => <div data-testid="Breadcrumb">{breadcrumbPairs && breadcrumbPairs[0]?.title}</div>);
jest.mock('@components/magic/CustomTittle/CustomTitle', () => ({ title }) => <div data-testid="CustomTittle">{title}</div>);

describe('Vaccination Campaigns page', () => {
  it('renders all main sections and child components', () => {
    render(<Campaigns />);
    expect(screen.getByTestId('Breadcrumb')).toBeInTheDocument();
    expect(screen.getByTestId('CustomTittle')).toHaveTextContent('Vaccination Campaign');
    expect(screen.getByTestId('LatestVaccinationCampaign')).toBeInTheDocument();
  });

  it('renders the correct breadcrumb titles', () => {
    render(<Campaigns />);
    expect(screen.getByTestId('Breadcrumb')).toHaveTextContent('Dashboard');
  });

  it('applies the correct styles to the root div', () => {
    const { container } = render(<Campaigns />);
    const rootDiv = container.firstChild;
    expect(rootDiv).toHaveStyle('background: #E6F8F9');
    expect(rootDiv).toHaveStyle('padding-top: 10px');
    expect(rootDiv).toHaveStyle('padding-bottom: 50px');
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Campaigns />);
    expect(asFragment()).toMatchSnapshot();
  });
});
