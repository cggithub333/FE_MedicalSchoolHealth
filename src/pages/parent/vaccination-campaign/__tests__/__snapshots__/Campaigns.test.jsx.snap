// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Vaccination Campaigns page matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="background: rgb(230, 248, 249); height: 100vh; padding-bottom: 50px; padding-top: 10px;"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row css-d64g0g-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-6 css-1ml5fja-MuiGrid-root"
      >
        <div
          data-testid="Breadcrumb"
        >
          Dashboard
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row css-d64g0g-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-6 css-hu5rp3-MuiGrid-root"
      >
        <div
          data-testid="CustomTittle"
        >
          Vaccination Campaign
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row css-yuvp34-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-10 css-r13beu-MuiGrid-root"
      >
        <div
          data-testid="LatestVaccinationCampaign"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
